spec/acts_as_taggable_on/tag_spec.rb

```diff
@@ -167,6 +167,17 @@
     it 'should return an empty array if no tags are specified' do
       expect(ActsAsTaggableOn::Tag.find_or_create_all_with_like_by_name([])).to be_empty
     end
+
+    context 'retry 3 times on not unique exception' do
+      it 'performs 3 tries before raising the exception' do
+        allow(ActsAsTaggableOn::Tag).to receive(:named_any).and_raise(ActiveRecord::RecordNotUnique.new('error')) # trigger error inside block
+        expect(ActiveRecord::Base.connection).to receive(:execute).with('ROLLBACK').exactly(3).times
```

Comment:
expect(ActiveRecord::Base.connection).to receive(:execute).with('ROLLBACK').exactly(4).times

Datacleaner will rollback too.

Without datacleaner this test will rollback only 3 time.